sdk/servicebus/azure-servicebus/migration_guide.md

```diff
@@ -0,0 +1,149 @@
+# Guide to migrate from azure-servicebus v0.50 to v7
+
+This document is intended for users that are familiar with v0.50 of the Python SDK for Service Bus library (`azure-servicebus 0.50.x`) and wish 
+to migrate their application to v7 of the same library.
+
+For users new to the Python SDK for Service Bus, please see the [readme file for the azure-servicebus](./README.md).
+
+## General changes
+Version 7 of the azure-servicebus package is the result of our efforts to create a client library that is user-friendly and idiomatic to the Python ecosystem.
+Alongside an API redesign driven by the new [Azure SDK Design Guidelines for Python](https://azure.github.io/azure-sdk/python_introduction.html#design-principles), 
+the latest version improves on several areas from v0.50.
+
+Note: The large version gap is in order to normalize service bus SDK versions across our languages, as they consolidate on structure as well.
+
+### Specific clients for sending and receiving
+In v7 we've simplified the API surface, making two distinct clients, rather than one for each of queue, topic, and subscription:
+* `ServiceBusSender` for sending messages. [Sync API](https://azuresdkdocs.blob.core.windows.net/$web/python/azure-servicebus/1.0.0/azure.servicebus.html#azure.eventhub.ServiceBusSender)
+and [Async API](https://azuresdkdocs.blob.core.windows.net/$web/python/azure-servicebus/1.0.0/azure.servicebus.aio.html#azure.servicebus.aio.ServiceBusSender)
+* `ServiceBusReceiver` for receiving messages. [Sync API](https://azuresdkdocs.blob.core.windows.net/$web/python/azure-servicebus/1.0.0/azure.servicebus.html#azure.eventhub.ServiceBusReceiver)
+and [Async API](https://azuresdkdocs.blob.core.windows.net/$web/python/azure-servicebus/1.0.0/azure.servicebus.aio.html#azure.servicebus.aio.ServiceBusReceiver)
+
+As a user this will be largely transparent to you, as initialization will still occur primarily via the top level ServiceBusClient,
+the primary difference will be that rather than creating a queue_client, for instance, and then a sender off of that, you would simply
+create a servicebus queue sender off of your ServiceBusClient instance via the "get_queue_sender" method.
+
+It should also be noted that many of the helper methods that previously existed on the intermediary client (e.g. QueueClient and Peek) now
+exist on the receiver (in the case of peek) or sender itself.  This is to better consolidate functionality and align messaging link lifetime
+semantics with the sender or receiver lifetime.
+
+### Client constructors
+
+| In v7 | Equivalent in v5 | Sample |
+|---|---|---|
+| `ServiceBusClient.from_connection_string()`    | `ServiceBusClient.from_connection_string()` | [using credential](./samples/sync_samples/sample_code_servicebus.py ) |
+| `QueueClient.from_connection_string()`    | `ServiceBusClient.from_connection_string().get_queue_<sender/receiver>()` | [client initialization](./samples/sync_samples/send_queue.py ) |
+
+### Receiving events 
```

Comment:
nit: events -> messages

We should use "messages" instead of "events" :)

Reply:
updated, thanks for call out